apps/blestress/src/rx_stress.c

```diff
@@ -0,0 +1,876 @@
+        .cb = rx_stress_7_gap_event,
+        .pattern_data = NULL,
+        .pattern_len = 0,
+    },
+    {
+        .instance = 1,
+        .instance_uuid128 = rx_stress_8_uuid128,
+        .legacy_pdu = EXTENDED_ADVERT,
+        .cb = rx_stress_8_gap_event,
+        .pattern_data = NULL,
+        .pattern_len = 0,
+    },
+    {
+        .instance = 1,
+        .instance_uuid128 = rx_stress_9_uuid128,
+        .legacy_pdu = EXTENDED_ADVERT,
+        .cb = rx_stress_9_gap_event,
+        .pattern_data = NULL,
+        .pattern_len = 0,
+    },
+};
+
+static void
+rx_stress_9_perform()
+{
+    do {
+        /* Test is based on gap event handling, so start advert. */
+        rx_stress_simple_adv(rx_stress_adv_sets + 9);
+        os_sem_pend(&rx_stress_main_sem, OS_TIMEOUT_NEVER);
+        /* Stop test when TX device has terminated all connections or
+         * number of connections has reached the max possible value. */
+    } while (rx_stress_ctxs[9].con_num != 0 &&
+             rx_stress_ctxs[9].con_num < MYNEWT_VAL(BLE_MAX_CONNECTIONS) + 1);
+
+    rx_stress_on_test_finish(9);
+    //console_printf("\033[0;32m\nStress test 9 completed\033[0m\n");
+}
+
+static void
+rx_stress_start(int test_num)
+{
+    /* Init semaphore with 0 tokens. */
+    os_sem_init(&rx_stress_main_sem, 0);
+
+    console_printf("\033[1;36mStart test num %d\033[0m\n", test_num);
+
+    /* Start test. */
+    switch (test_num) {
+    case 2:
+        rx_stress_simple_adv(rx_stress_adv_sets + 2);
+        break;
+    case 3:
+        rx_stress_simple_adv(rx_stress_adv_sets + 3);
+        break;
+    case 4:
+        rx_stress_simple_adv(rx_stress_adv_sets + 4);
+        break;
+    case 5:
+        rx_stress_simple_adv(rx_stress_adv_sets + 5);
+        break;
+    case 6:
+        rx_stress_simple_adv(rx_stress_adv_sets + 6);
+        break;
+    case 7:
+        rx_stress_simple_adv(rx_stress_adv_sets + 7);
+        break;
+    case 8:
+        rx_stress_simple_adv(rx_stress_adv_sets + 8);
+        break;
+    case 9:
+        rx_stress_9_perform();
+        break;
+    default:
+        console_printf("\033[0;31mFound test, but do not know how to perform."
+                       "\033[0m\n");
+        return;
+    }
+
+    /* Wait for the test to finish. Then 1 token will be released
+    * allowing to pass through semaphore. */
+    os_sem_pend(&rx_stress_main_sem, OS_TIMEOUT_NEVER);
+}
+
+static void
+rx_stress_main_task_fn(void *arg)
+{
+    int i;
+    console_printf("\033[1;36mRX device\033[0m\n");
+
+    /* Start SWITCH advert that gives possibility to remotely start
+     * next test advert */
+    rx_stress_simple_adv(rx_stress_adv_sets);
+
+    for (i = 2; i <= 9; ++i) {
+        /* Start test. */
+        rx_stress_start(i);
+    }
+
+    /* Print tests results */
+    com_stress_print_report(rx_stress_ctxs);
+
+    /* Task should never return */
+    while (1) {
+        os_time_delay(os_time_ms_to_ticks32(OS_TIME_MAX));
```

Comment:
why this delay is needed here ? isn't while (1) enough ? - what is asm difference ? - how looks while(1) ?